chain/client/src/config_updater.rs

```diff
@@ -10,25 +11,42 @@ pub struct ConfigUpdater {
 
     /// Represents the latest Error of reading the dynamically reloadable configs.
     updateable_configs_error: Option<Arc<UpdateableConfigLoaderError>>,
+    /// Represents whether validator key was updated during the last reload.
+    validator_signer_updated: bool,
 }
 
 impl ConfigUpdater {
     pub fn new(
         rx_config_update: Receiver<Result<UpdateableConfigs, Arc<UpdateableConfigLoaderError>>>,
     ) -> Self {
-        Self { rx_config_update, updateable_configs_error: None }
+        Self { rx_config_update, updateable_configs_error: None, validator_signer_updated: false }
+    }
+
+    pub fn was_validator_signer_updated(&self) -> bool {
+        self.validator_signer_updated
     }
 
     /// Check if any of the configs were updated.
     /// If they did, the receiver (rx_config_update) will contain a clone of the new configs.
-    pub fn try_update(&mut self, update_client_config_fn: &dyn Fn(UpdateableClientConfig)) {
+    pub fn try_update(
+        &mut self,
+        update_client_config_fn: &dyn Fn(UpdateableClientConfig),
+        update_validator_signer_fn: &dyn Fn(Arc<ValidatorSigner>) -> bool,
```

Comment:
what does this bool return value represent? I could not find where it is generated and returned.

Reply:
It is return type of `update_validator_signer_fn` argument.